src/libsyntax/feature_gate.rs

```diff
@@ -385,6 +386,13 @@ impl<'a> Context<'a> {
                 return;
             }
         }
+        for &(ref n, ref ty) in self.plugin_attributes.iter() {
+            if &*n == name {
+                // Plugins can't gate attributes, so we don't check for it
```

Comment:
Oh, so the comment is just contrasting to the behaviour of the code above. Makes sense. 

Could you still point out that the loop is still necessary to short-circuit to avoid the checks below.